components/script/dom/htmlfieldsetelement.rs

```diff
@@ -186,33 +188,65 @@ impl VirtualMethods for HTMLFieldSetElement {
                 let fields = children.flat_map(|child| {
                     child
                         .traverse_preorder(ShadowIncluding::No)
-                        .filter(|descendant| {
-                            matches!(
-                                descendant.type_id(),
-                                NodeTypeId::Element(ElementTypeId::HTMLElement(
-                                    HTMLElementTypeId::HTMLButtonElement,
-                                )) | NodeTypeId::Element(ElementTypeId::HTMLElement(
-                                    HTMLElementTypeId::HTMLInputElement,
-                                )) | NodeTypeId::Element(ElementTypeId::HTMLElement(
-                                    HTMLElementTypeId::HTMLSelectElement,
-                                )) | NodeTypeId::Element(ElementTypeId::HTMLElement(
-                                    HTMLElementTypeId::HTMLTextAreaElement,
-                                ))
-                            )
+                        .filter(|descendant| match descendant.type_id() {
+                            NodeTypeId::Element(ElementTypeId::HTMLElement(
+                                HTMLElementTypeId::HTMLButtonElement,
+                            )) |
+                            NodeTypeId::Element(ElementTypeId::HTMLElement(
+                                HTMLElementTypeId::HTMLInputElement,
+                            )) |
+                            NodeTypeId::Element(ElementTypeId::HTMLElement(
+                                HTMLElementTypeId::HTMLSelectElement,
+                            )) |
+                            NodeTypeId::Element(ElementTypeId::HTMLElement(
+                                HTMLElementTypeId::HTMLTextAreaElement,
+                            )) => true,
+                            NodeTypeId::Element(ElementTypeId::HTMLElement(
+                                HTMLElementTypeId::HTMLElement,
+                            )) => descendant
+                                .downcast::<HTMLElement>()
+                                .unwrap()
+                                .is_form_associated_custom_element(),
+                            _ => false,
                         })
                 });
                 if disabled_state {
                     for field in fields {
                         let el = field.downcast::<Element>().unwrap();
```

Comment:
While you're here. :)

```suggestion
                        let element = field.downcast::<Element>().unwrap();
```